components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/deployment/config/deployer/ConfigDeployer.java

```diff
@@ -0,0 +1,252 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+
+public class ConfigDeployer implements AppDeploymentHandler {
+
+    private static final Log log = LogFactory.getLog(ConfigDeployer.class);
+
+    private static final String PROPERTY_TYPE = "config/property";
+
+    private static final String LOCAL_CONFIG_FILE_NAME = "config.properties";
+    private static final String GLOBAL_CONFIG_FILE_NAME = "file.properties";
+    private static final String IS_CERT_DEPLOYMENT_ENABLED = "isCertDeploymentEnabled";
+    private Properties globalProperties;
+
+    public static final char URL_SEPARATOR_CHAR = '/';
+
+    public ConfigDeployer() {
+    }
+
+    @Override
+    public void deployArtifacts(CarbonApplication carbonApp, AxisConfiguration axisConfig) throws DeploymentException {
+        if (log.isDebugEnabled()) {
+            log.debug("Deploying properties  - " + carbonApp.getAppName());
+        }
+        ApplicationConfiguration appConfig = carbonApp.getAppConfig();
+        List<Artifact.Dependency> deps = appConfig.getApplicationArtifact().getDependencies();
+
+        List<Artifact> artifacts = new ArrayList<Artifact>();
+        for (Artifact.Dependency dep : deps) {
+            if (dep.getArtifact() != null) {
+                artifacts.add(dep.getArtifact());
+            }
+        }
+        deployConfigArtifacts(artifacts, carbonApp.getAppNameWithVersion());
+    }
+
+    @Override
+    public void undeployArtifacts(CarbonApplication carbonApp, AxisConfiguration axisConfig) throws DeploymentException {
+
+    }
+
+    private void deployConfigArtifacts(List<Artifact> artifacts, String parentAppName) {
+        artifacts.stream().filter(artifact -> PROPERTY_TYPE.equals(artifact.getType())).forEach(artifact -> {
+            if (log.isDebugEnabled()) {
+                log.debug("Deploying config artifact: " + artifact.getName());
+            }
+            writePropertyToMap(artifact);
+        });
+    }
+
+    private void writePropertyToMap(Artifact artifact) {
+        // get the file path of the registry config file
+        List<CappFile> files = artifact.getFiles();
+        if (files.size() == 1) {
+            Path confFolder = Paths.get(getHome(), "conf");
+            Path globalPropertiesFilePath = confFolder.resolve(GLOBAL_CONFIG_FILE_NAME) ;
+            Path serverConfPropertyPath = confFolder.resolve(LOCAL_CONFIG_FILE_NAME);
+            String configFilePath = artifact.getExtractedPath() + File.separator + LOCAL_CONFIG_FILE_NAME;
+            processConfFile(configFilePath, globalPropertiesFilePath.toString(), serverConfPropertyPath.toString());
+        } else {
+            log.error("config/property type must have a single file which declares " +
+                    "config. But " + files.size() + " files found.");
+        }
+    }
+
+    public void processConfFile(String configFilePath, String globalPropertiesFilePath, String serverConfPropertyPath) {
```

Comment:
can this be private

Reply:
When I discussed this with Danesh aiya, he suggested writing a separate method to add the config and cert. He mentioned that if it's needed in his feature, he can use it, so I made it public.

@daneshk Shall I make it private instead?

Reply:
Earlier, I thought of calling this in the docker build time to import certificate at build time. But we need to enable certificate import at deployment time as well. So calling this in build time doesn't work. Let's make with private. we may need to allow cert import at server startup.